cdap-app-templates/cdap-etl/cdap-etl-core/src/test/java/co/cask/cdap/etl/common/TransformExecutorTest.java

```diff
@@ -56,30 +58,35 @@ public void testEmptyTransforms() throws Exception {
   @Test
   public void testTransforms() throws Exception {
     MockMetrics mockMetrics = new MockMetrics();
+    DataTracerFactory dataTracerFactory = new NoopDataTracerFactory();
     Map<String, TransformDetail> transformationMap = new HashMap<>();
 
     transformationMap.put("transform1",
                           new TransformDetail(
                             new TrackedTransform<>(new IntToDouble(),
-                                                   new DefaultStageMetrics(mockMetrics, "transform1")),
+                                                   new DefaultStageMetrics(mockMetrics, "transform1"),
+                                                   dataTracerFactory.getDataTracer(null, "transform1")),
                             ImmutableList.of("transform2", "sink1")));
 
     transformationMap.put("transform2",
                           new TransformDetail(
                             new TrackedTransform<>(new Filter(100d, Threshold.LOWER),
-                                                   new DefaultStageMetrics(mockMetrics, "transform2")),
+                                                   new DefaultStageMetrics(mockMetrics, "transform2"),
+                                                   dataTracerFactory.getDataTracer(null, "transform2")),
```

Comment:
why pass `null` for application id? Even if its not used better to pass the `new ApplicationId()`. Also should we validate that the tracer name is non-null?